### DISCUSSION TITLE
Popup/Redirect possible for Custom Wallet connector?

### DISCUSSION THREAD

**Discussion by chimalsky** [wevm/wagmi#2174] [Q&A]
I'm creating a custom-connector that requires a user to be redirected to a separate webpage. Is this within the scope of a wagmi custom connector?

The closest existing metaphor would be Oauth Google signon. When an app allows you to signon with the Google Signon, you are either redirected to the Google Signon page or you are shown a popup with the Google Signon UI.

Does WAGMI prohibit such a user-interaction implementation within a custom connector? It seems like all existing examples leverage a browser extension (metamask for example)

**Comment by tmm** [comment]
> Does WAGMI prohibit such a user-interaction implementation within a custom connector?

Nope, you can do pretty much whatever you want.

> It seems like all existing examples leverage a browser extension (metamask for example)

Some other Connector "types" worth checking out:

- iframe https://github.com/wagmi-dev/references/blob/main/packages/connectors/src/safe.ts
- Third-party SDK with interface https://github.com/wagmi-dev/references/blob/main/packages/connectors/src/walletConnect.ts
- Social auth https://github.com/EveripediaNetwork/wagmi-magic-connector

**Reply by chimalsky** [reply]
Thank you, @tmm !
